src/error.rs

```diff
@@ -70,7 +70,7 @@ impl error::Error for Error {
         }
     }
 
-    fn cause(&self) -> Option<&error::Error> {
+    fn cause(&self) -> Option<&dyn error::Error> {
```

Comment:
this may not have been a great idea if the plan is to support old versions of Rust